Inc/HALALMock/Services/SharedMemory/SharedMemory.hpp

```diff
@@ -10,8 +10,16 @@ class SharedMemory {
 
     // do not use before SharedMemory::start, uninitialized!!!
     static EmulatedPin *gpio_memory;
+	static void* emulated_state_machine{};
```

Comment:
you don't need to keep this variable around, just static_cast the result from mmap inside the function